include/alpaka/dev/DevOmp5.hpp

```diff
@@ -0,0 +1,365 @@
+                }
+            };
+        }
+    }
+    namespace mem
+    {
+        namespace buf
+        {
+            template<
+                typename TElem,
+                typename TDim,
+                typename TIdx>
+            class BufOmp5;
+
+            namespace traits
+            {
+                //#############################################################################
+                //! The CUDA RT device memory buffer type trait specialization.
+                template<
+                    typename TElem,
+                    typename TDim,
+                    typename TIdx>
+                struct BufType<
+                    dev::DevOmp5,
+                    TElem,
+                    TDim,
+                    TIdx>
+                {
+                    using type = mem::buf::BufOmp5<TElem, TDim, TIdx>;
+                };
+            }
+        }
+    }
+    namespace pltf
+    {
+        namespace traits
+        {
+            //#############################################################################
+            //! The CUDA RT device platform type trait specialization.
+            template<>
+            struct PltfType<
+                dev::DevOmp5>
+            {
+                using type = pltf::PltfOmp5;
+            };
+        }
+    }
+    namespace queue
+    {
+        using QueueOmp5NonBlocking = QueueGenericThreadsNonBlocking<dev::DevOmp5>;
+        using QueueOmp5Blocking = QueueGenericThreadsBlocking<dev::DevOmp5>;
+
+        namespace traits
+        {
+            template<>
+            struct QueueType<
+                dev::DevOmp5,
+                queue::Blocking
+            >
+            {
+                using type = queue::QueueOmp5Blocking;
+            };
+
+            template<>
+            struct QueueType<
+                dev::DevOmp5,
+                queue::NonBlocking
+            >
+            {
+                using type = queue::QueueOmp5NonBlocking;
+            };
+        }
+    }
+    namespace wait
+    {
+        namespace traits
+        {
+            //#############################################################################
+            //! The thread Omp5 device wait specialization.
+            //!
+            //! Blocks until the device has completed all preceding requested tasks.
+            //! Tasks that are enqueued or queues that are created after this call is made are not waited for.
+            template<>
+            struct CurrentThreadWaitFor<
+                dev::DevOmp5>
+            {
+                //-----------------------------------------------------------------------------
+                ALPAKA_FN_HOST static auto currentThreadWaitFor(
+                    dev::DevOmp5 const & dev)
+                -> void
+                {
+                    ALPAKA_DEBUG_FULL_LOG_SCOPE;
+
+                    generic::currentThreadWaitForDevice(dev);
+// #pragma omp taskwait
```

Comment:
?

Reply:
Either this is left over from the old OpenMP4, or I was thinking about implementing an OpenMP tesking-based queue. Removed.